ansible/roles/beegfs/tasks/main.yml

```diff
@@ -0,0 +1,72 @@
+---
+- name: Ensure the BeeGFS package repo is defined
+  get_url:
+    url: "{{ beegfs_repo_url }}"
+    dest: /etc/yum.repos.d/beegfs-rhel7.repo
+  become: True
+
+- name: Ensure any required kernel modules are loaded
+  modprobe:
+    name: "{{ item }}"
+    state: present
+  become: True
+  with_items: "{{ beegfs_kmod_preload }}"
+
+- include: mgmt.yml
+  when: beegfs_enable.mgmt
+
+- include: admon.yml
+  when: beegfs_enable.admon
+
+- include: meta.yml
+  when: beegfs_enable.meta
+
+- include: oss.yml
+  when: beegfs_enable.oss
+
+- include: client.yml
+  when: beegfs_enable.client
```

Comment:
I think it might be simpler to create a separate role for the client. Just thinking about using this for spark, I am likely just to have the client. Or am I missing the point of how you are laying this out, maybe its more hyper-converged than that?

Reply:
Using the role to create clients for an existing filesystem, I'm not sure whether that will work as-is.  There's a little setup in common, not too much, so we could feasibly split it out.